### PR TITLE
[Android] Add support for translucent header

### PR DESCRIPTION
**Pull request by andreibarabas**
Currently, the `headerTranslucent = true` is only supported on iOS. With this PR the setting will also work on Android

### CODE REVIEW

**Comment by kyle-ssg** [comment]
Ah awesome! I Guess that would resolve https://github.com/software-mansion/react-native-screens/issues/573.

**Comment by kyle-ssg** [comment]
@andreibarabas This works exactly as I'd expect except when popping, it seems to force everything to render after the navbar.

**Comment by andreibarabas** [comment]
Hey @kyle-ssg - i'll take a look

**Comment by WoLewicki** [comment]
I would love to merge it! Please inform me when it is ready for review. It should also close #405.

**Comment by andreibarabas** [comment]
Hey @WoLewicki , I need a couple more days to get to this. I'm currently caught up in a release, and once that is done can look more into this.

File: android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java [hunk 1]
```diff
@@ -76,7 +77,6 @@ public ScreenStackHeaderConfig(Context context) {
     if (context.getTheme().resolveAttribute(android.R.attr.colorPrimary, tv, true)) {
       mToolbar.setBackgroundColor(tv.data);
     }
-    mToolbar.setClipChildren(false);
```
Comment (janicduplessis) [inline android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:79]: Is there a reason why this was removed?

Comment (WoLewicki) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:79]: Probably wrongly rebased to the current master.

**Comment by WoLewicki** [comment]
@andreibarabas it is not a problem 🙂

**Comment by WoLewicki** [comment]
How are things going on this PR @andreibarabas? Do you need any help or you just didn't have time to finish it?

**Comment by andreibarabas** [comment]
Haven't yet had the time to dive deep into it, but will give it another shoot hopefully tomorrow / the day after that.

I also need this fixed for our production build, so top priority ;)

**Comment by andreibarabas** [comment]
@WoLewicki - done ;) it works now, tested it with pop also.

**Comment by WoLewicki** [comment]
Looks good! There are some other places where the information about the platform availability should be changed: https://github.com/software-mansion/react-native-screens/blob/master/README.md#L194, https://github.com/software-mansion/react-native-screens/blob/master/src/index.d.ts#L158, https://github.com/software-mansion/react-native-screens/blob/master/src/native-stack/types.tsx#L116. If you could change this, I would be grateful. Also, do you know of any options to have something like `contentInsetAdjustmentBehavior` for Android? Would be good to have consistent behavior for ScrollViews on both platforms.

**Comment by kyle-ssg** [comment]
Hey @andreibarabas, from some brief testing I've noticed the title isn't rendering at least for me.

image

**Comment by andreibarabas** [comment]
Hey @kyle-ssg . Title i have't checked. Our design system doesn't use them.

Let me check that and get back.

Any other issues?

**Comment by kyle-ssg** [comment]
> Hey @kyle-ssg . Title i have't checked. Our design system doesn't use them.
> 
> Let me check that and get back.
> 
> Any other issues?

Other than that it looks awesome @andreibarabas!

**Comment by andreibarabas** [comment]
@kyle-ssg - can you provide a snack maybe? I checked on our app and put a dummy title and it showed up. Here it is 
Screenshot_20200915_094510

@WoLewicki - updated the doc as per your suggestion. 

As for `contentInsetAdjustmentBehavior`, I'm not really an android developer, so would have to deeply research it. maybe in a different PR? Do you see that as blocking for this one?

**Comment by WoLewicki** [comment]
@andreibarabas I don't think it is blocking since this transparent header is an optional thing anyway.

**Comment by kyle-ssg** [comment]
Yeah, happy with that. I'll raise separately with repro

File: android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java [hunk 2]
```diff
@@ -363,7 +367,7 @@ public void setTintColor(int color) {
 
   public void setTopInsetEnabled(boolean topInsetEnabled) { mIsTopInsetEnabled = topInsetEnabled; }
 
-  public void setBackgroundColor(int color) {
+  public void setBackgroundColor(Integer color) {
```
Comment (WoLewicki) [inline android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:370]: Also, could you change it back to int? It is the way we pass all the color props and I think it is the preferred way of doing so.

Comment (andreibarabas) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:370]: It was int, but int does not accept null. We need null to actually not put a color.

Comment (andreibarabas) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:370]: For the ui to look as in the example, on iOS (and now android), you need to write


`headerTranslucent: true,
      headerStyle: {backgroundColor: 'transparent'},`

Comment (WoLewicki) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:370]: If you make it an Integer, the value of 0 is passed for `backgroundColor: 'transparent',` and -1 if nothing is passed. There is no `null` option. Am I right?
If you make it an int, the same values are passed. 
So maybe it is better to leave the int and just check `if (mBackgroundColor != -1)` instead of 0?

Comment (andreibarabas) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:370]: let me quickly do a test and see how it behaves

Comment (andreibarabas) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java]: @WoLewicki - you where right. I think I put null because it clearly says what you are doing (i.e: setting the color ONLY if the prop was set), but your solution works and i've reverted it like that to be consistent.

Here are some screenshots:

Screenshot_20200915_150323
Screenshot_20200915_150357
Screenshot_20200915_150423
Screenshot_20200915_150446

File: README.md [hunk 3]
```diff
@@ -191,9 +191,9 @@ String that applies `rtl` or `ltr` form to the stack.
 
 When set to `false` the back swipe gesture will be disabled when the parent `Screen` is on top of the stack. The default value is `true`.
 
-#### `translucent` (iOS only)
+#### `translucent`
 
-When set to `true`, it makes native navigation bar on iOS semi-transparent with blur effect. It is a common way of presenting a navigation bar introduced in iOS 11. The default value is `false`.
+When set to `true`, it makes native navigation bar on iOS semi-transparent with blur effect, while on Android is fully transparent. It is a common way of presenting a navigation bar introduced in iOS 11. The default value is `false`.
```
Comment (WoLewicki) [inline README.md:196]: I also think this is not true for either of the platforms. All it does is make it possible for the content to be visible under the header due to not rendering content below the header but under it. It does not change the transparency by itself since it is controlled by `backgroundColor`. Or am I missing something?

Comment (andreibarabas) [reply README.md]: Yes, you are right. Updated.

**Comment by andreibarabas** [comment]
hey @WoLewicki  - anything else you guys need to have this merged?

**Comment by WoLewicki** [comment]
Didn't @kyle-ssg have problems with the title that needed to be resolved? Or are they gone?

**Comment by andreibarabas** [comment]
I can't repro them. All screenshots I sent above include the title

**Comment by brunocrpontes** [comment]
I tested this branch in a personal app and headers title just go away when use an headerLeft component in bar.

**Comment by WoLewicki** [comment]
@brunocrpontes is it specific to translucent header? I think you are talking about this behavior: https://github.com/software-mansion/react-native-screens/tree/master/native-stack#headerleft and the same happens with normal header. Am I right?

**Comment by brunocrpontes** [comment]
> @brunocrpontes is it specific to translucent header? I think you are talking about this behavior: https://github.com/software-mansion/react-native-screens/tree/master/native-stack#headerleft and the same happens with normal header. Am I right?

Yes, you're right.

**Comment by WoLewicki** [comment]
@kyle-ssg so can you provide a repro showing the lack of title? I would like to test it and merge it if all is ok.

**Comment by kyle-ssg** [comment]
Hey @WoLewicki sorry today I can't really busy release, it would make sense if it's the custom header left that's doing it though. I bet that is the usecase!

File: android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java [hunk 4]
```diff
@@ -238,7 +242,7 @@ public void onUpdate() {
     }
 
     // background
-    if (mBackgroundColor != 0) {
+    if (mBackgroundColor != -1) {
```
Comment (kmagiera) [inline android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:245]: Can you explain why this change is made here? If we expect "default" to be `-1` maybe we should also set the default in `@ReactProp` while defining bgColor:

```java
@ReactProp(name = "backgroundColor", customType = "Color", defaultInt = -1)
```

??

**Review by kmagiera — COMMENTED** [review]
This looks good overal, I only have one question I left inline regarding one change. I'd like to get that one item clarified, and this is good to merge otherwise

File: android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java (hunk 4 above)
Comment (WoLewicki) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:245]: I think it is a good idea to add that for extra safety.

Comment (andreibarabas) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:245]: Sure. will take care of it as soon as possible.

There is still an open issue from @kyle-ssg which reproduces only if there is a custom left component. I won't have time earlier than next week to investigate

Comment (WoLewicki) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:245]: But I think this is not an issue since it is the expected behavior for the title to disappear when there is `headerLeft` component.

Comment (kyle-ssg) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:245]: Regardless of if the headerLeft occurred before / after this work I think this is good to merge, I'll look into this separately and raise something if I think there's an issue. @WoLewicki @andreibarabas.

Comment (WoLewicki) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:245]: On iOS, the situation is kind of similar but for `headerCenter`, where using it makes the title disappear.

Comment (andreibarabas) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:245]: sure. it's fine with me. I added @kmagiera suggestion, so let me know if you need anything else from my side to get this merged.

**Comment by WoLewicki** [comment]
Thanks for your contribution! The only problem I can spot is that passing a wrong color name is resolved to 0 so the header becomes transparent, but I think we can classify it as a problem of the programmer passing wrong options. I am merging it then!

File: android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfigViewManager.java [hunk 5]
```diff
@@ -89,9 +89,9 @@ public void setTitleColor(ScreenStackHeaderConfig config, int titleColor) {
     config.setTitleColor(titleColor);
   }
 
-  @ReactProp(name = "backgroundColor", customType = "Color")
-  public void setBackgroundColor(ScreenStackHeaderConfig config, int titleColor) {
-    config.setBackgroundColor(titleColor);
+  @ReactProp(name = "backgroundColor", customType = "Color", defaultInt = -1)
```
Comment (andrewbeckman) [inline android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfigViewManager.java:92]: @andreibarabas I am noticing that when I try to set my header background color to "#fff" or "#ffffff" (i.e. white), I end up with a non-white header. If you take a look at the attached photo, you can see the header appears to be gray. Also note that the status bar at the top is white.

Any chance these bugs came in with the changes to this file?

Screen Shot 2020-10-04 at 12 12 00 PM

Comment (andrewbeckman) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfigViewManager.java:92]: To clarify, this is when `headerTranslucent` is set to false.

Comment (andrewbeckman) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfigViewManager.java:92]: After some further testing, @andreibarabas it looks like it was the change in ScreenStackHeaderConfig.java

File: android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java (hunk 4 above)
Comment (andrewbeckman) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java:245]: I am testing these changes out and this line seems to have introduced a bug where I can't set the header background color to "#fff" or "#ffffff".

File: android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfigViewManager.java (hunk 5 above)
Comment (WoLewicki) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfigViewManager.java:92]: Looks like the "#fff", "#ffffff" and `white` colors all have value of "-1" on the native side, so the newly added if makes it impossible to use these values. Can you check if https://github.com/software-mansion/react-native-screens/pull/656 fixes the issue?

Comment (andrewbeckman) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfigViewManager.java:92]: @WoLewicki that fixed the issue! Thanks

Comment (andreibarabas) [reply android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfigViewManager.java:92]: Guys, glad you solved it. Thanks! Just now had the time to groom the notifications.
